docs/apis/amr-api-reference.md

```diff
@@ -0,0 +1,264 @@
+
+##### IncludeExtendedMetadata
+
+Optional.
+
+Default value is `false`.
+
+When set to `false`, AMR API reads basic metadata:
+
+- List
+- Folder
+- File
+- List Item
+- Roles
+- Role Assignments
+
+When set to `true`,  AMR API reads all metadata available:
+
+For Files:
+
+- Web Part
+- Web Part personalization
+- Links
+- Version events
+- Event receivers
+- Attachment metadata
+
+For Lists:
+
+- Custom actions
+- List shortcuts
+
+For List Items:
+
+- Comments
+- Documents set links
+- Activities
+- List Item shortcuts
+
+Including extended metadata slows down the read significantly. For file share migrations, keep the default value `false`. Set to `true` only when necessary, for complex migration projects.
+
+##### IncludePermission
+
+Optional.
+
+A **Bool** value to indicate if permissions read is needed. Default value is `false`.
+
+When set to `true`, AMR API reads permission metadata in `RoleAssignments` tags in `Manifest.xml` files. The file includes all distinguished permission metadata for each read SharePoint object, along with property `ScopeId`.
+
+##### Using IncludeSecurity with IncludePermission
+
+`IncludeSecurity` reads the full list of all Users and Groups on the target Site. Optionally use `IncludePermission` to read permission metadata, if needed. There are four possible combinations for using both **readOption**:
+
+- `IncludeSecurity`=`true`, `IncludePermission`=`false`: reads the full list of all Users and Groups information of the target Site, without permissions.
+- `IncludeSecurity`=`false`, `IncludePermission`=`true`: reads permissions with a list of corresponding User and Group information, related to the permissions read.
+- `IncludeSecurity`=`true`, `IncludePermission`=`true`: reads permissions with the full list of all User and Group information of the target Site.
+- `IncludeSecurity`=`false`, `IncludePermission`=`false`: reads no permissions, Users or Groups information.
+
+##### StartChangeToken
+
+Optional.
+
+A **Integer** value containing the changeToken item.
+
+By default, when no `StartChangeToken` is provided, `CreateSPAsyncReadJob` method returns all items available, based on the parameters. A `CurrentChangeToken` value is returned every time.
+
+To read only the items that changed since last read, set a `StartChangeToken` in subsequent calls to `CreateSPAsyncReadJob`. Use `CurrentChangeToken` returned from last call as the value of `StartChangeToken`.
+
+AMR API returns an error and stops the read, if it receives an invalid `StartChangeToken` value.
+
+Be careful when using this feature with large number of items. The read job could run for extended duration. AMR API cancels jobs that run over 10 minutes to protect the SharePoint infrastructure.
+
+#### encryptionOption
+
+Optional.
+
+A `EncryptionOption` object, containing the AES256CBCKey used to decrypt the output.
```

Comment:
Is 'AES256CBCKey' correct? should this be AES-256-CBC or AES256CBC? In other places, it looks like you used AES256CBC...